apps/src/code-studio/pd/form_components_func/FormController.jsx

```diff
@@ -76,6 +76,7 @@ const FormController = props => {
     serializeAdditionalData,
     sessionStorageKey,
     submitButtonText,
+    storage,
```

Comment:
Not sure this is necessary––it's helpful for consistent testing, but since storage should be sessionStorage, we don't need to pass it around like this

Reply:
you should also be able to stub sessionStorage on the window object, i.e. something like
```sinon.stub(window, 'sessionStorage', new FakeStorage())```
and get around all this extra routing

Reply:
(just make sure to clean it up afterwards so that it doesn't affect other tests)

Reply:
this PR suggests replacing window.sessionStorage is not possible: https://github.com/code-dot-org/code-dot-org/pull/34513 . if it does work, then go for it? fortunately, this only needs to work in whatever the latest version of ChromeHeadless is -- we don't have to worry about cross-browser compatibility for js unit tests.

Reply:
Very helpful, thanks!